### ISSUE TITLE
DataPackage constructor reuses internal structures

### ISSUE THREAD

**Issue by sbpcs59** [ropensci/datapackage#26]
To illustrate this
   - create a DataPackage and add a DataObject
  - list objects in the DataPackage
  - delete the DataPackage
  - create a new DataPackage
  - list objects in the new, empty DataPackage
  - the new DataPackage contains objects from the original, deleted package

To recreate
```
library(datapackage)
testdf <- data.frame(x=1:10,y=11:20)
csvfile <- tempfile(pattern = "file", tmpdir = tempdir(), fileext = ".csv")
write.csv(testdf, csvfile, row.names=FALSE)
cn <- CNode("SANDBOX")
mnId <- "urn:node:mnSandboxUCSB2"
dp <- new("DataPackage")
sciObj <- new("DataObject", format="text/csv", user="uid=slaughter,ou=Account,dc=ecoinformatics,dc=org", mnNodeId=mnId, filename=csvfile)
addData(dp, sciObj)
getIdentifiers(dp)
rm(dp)
newDP <- new("DataPackage")
getIdentifiers(newDP)
```

**Comment by mbjones** [comment]
@sbpcs59 yoiks. That's a problem!

**Comment by sbpcs59** [comment]
Fixed in commit eb0362146d259e3c3140ebf9a86f4224c761b880

The calls pkg <- DataPackage() and pkg <- new("DataPackage") now execute essentially
the same code path. Previously, pkg <- new("DataPackage") was not setting values for
the slots "@relations" or "@objects", so it appears with R garbage collection, these were
being reused (PS - I did test manual garbage collection i.e. gc()) which had no effect on
freeing these objects. So, the current initialize() sets new values and solves (covers up?)
the original problem.
